File: hermes.of.cheese.backend/src/main/java/com/sena/proyect/hermes/of/cheese/presentation/controller/request/DeliveryRequest.java
package com.sena.proyect.hermes.of.cheese.presentation.controller.request;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class DeliveryRequest {
    private String nameCustomer;
    private String addressCustomer;
    private String phoneCustomer;
    private List<DeliveryProductRequest> deliveryProductName = new ArrayList<>();

    @JsonIgnore
    private LocalDateTime dateDelivery = LocalDateTime.now();


}

File: HermesSenaProyect-frontend/hermes-frontend/src/app/model/delivery-request.ts
import { DeliveryProductRequest } from "./delivery-product-request";

export class DeliveryRequest {
    nameCustomer: string;
    addressCustomer: string;
    phoneCustomer: string;
    deliveryProductName: DeliveryProductRequest[];

    constructor(nameCustomer: string, adressCustomer: string, phoneCustomer: string, deliveryProducts: DeliveryProductRequest[]) {
        this.nameCustomer = nameCustomer;
        this.addressCustomer = adressCustomer;
        this.phoneCustomer = phoneCustomer;
        this.deliveryProductName = deliveryProducts;
    }
}
